### DISCUSSION TITLE
Out of Scope error with hdf5_fun.cpp

### DISCUSSION THREAD

**Discussion by mlangelaar** [gnudatalanguage/gdl#1517] [General]
Good afternoon,

Error in hdf5_cpp (out of scope), not sure how to fix this one, well versed in C, fair in C++

Scanning dependencies of target gdl
[ 15%] Building CXX object src/CMakeFiles/gdl.dir/hdf5_fun.cpp.o
/root/GDL/gdl/src/hdf5_fun.cpp: In destructor â?lib::hdf5_name_guard::~hdf5_name_guard()â?:
/root/GDL/gdl/src/hdf5_fun.cpp:113:44: error: â?H5free_memoryâ? was not declared in this scope
     ~hdf5_name_guard() { H5free_memory(name); }
                                            ^
make[2]: *** [src/CMakeFiles/gdl.dir/hdf5_fun.cpp.o] Error 1
make[1]: *** [src/CMakeFiles/gdl.dir/all] Error 2

Running on an older SL 7.8 O/S, have tried default compiler :
   gcc version 4.8.5 20150623 (Red Hat 4.8.5-39) (GCC) 

Also tried 'scl enable devtoolset-9 rh-python38 rh-perl530 bash'
  gcc version 9.3.1 20200408 (Red Hat 9.3.1-2) (GCC) 

The cmake is different, need to be able to call gdl from python, so :
   cd gdl/build
   cmake3 -DPYTHON=OFF -DPYTHON_MODULE=ON ..

Maiko

**Comment by slayoo** [comment]
Hello @mlangelaar 
The `H5free_memory` is part of the HDF5 API since version 1.8.13 (https://docs.hdfgroup.org/hdf5/v1_12/group___h5.html#title7).
Is it possible that your HDF5 library version is older?

**Reply by mlangelaar** [reply]
Hello Sylwester,
I thought I had checked the API version, but will do again, thanks for the specific info, much appreciated.
Maiko

**Reply by mlangelaar** [reply]
And of course I have hdf5 1.8.12, off by one :(
working on it, will let you (and discussion) know the outcome. thanks again.
Maiko

**Comment by mlangelaar** [comment]
Problem solved.
I was using HDF5 is 1.8.12 - should be minimum 1.8.13 (off by one sub sub version).
So I installed HDF5 from source, picking 1.10.8 out of the hat, and GDL compiles fine.
Thanks Sylwester - yes, the HDF5 API version was just slightly too old ...
Maiko
